deps/linenoise/linenoise.c

```diff
@@ -793,6 +793,31 @@ void linenoiseEditMoveRight(struct linenoiseState *l) {
     }
 }
 
+/* Consider letters/digits/underscore as “word”; others as delimiters. */
+static int isWordChar(char c) {
+    return (c == '_' || (c >= '0' && c <= '9') ||
+            (c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z'));
```

Comment:
What if there are some other special symbols in the middle, such as # @?
perhaps should just skip the space?